lib/shred.ts

```diff
@@ -230,25 +243,30 @@ function materializeRecordField(record, branch, rLevels, dLevel, value) {
       record[node.name] = record[node.name] || {};
 
       materializeRecordField(
-          record[node.name],
+          record[node.name] as Record<string,object[]>,
           branch.slice(1),
           rLevels,
           dLevel,
           value);
     }
   } else {
     if (node.repetitionType === "REPEATED") {
+      let recordArray = record[node.name] as object[]
       if (!(node.name in record)) {
         record[node.name] = [];
       }
 
-      while (record[node.name].length < rLevels[0] + 1) {
-        record[node.name].push(null);
+      while (recordArray.length < rLevels[0] + 1) {
+        recordArray.push({});
       }
 
-      record[node.name][rLevels[0]] = value;
+      recordArray[rLevels[0]] = value;
     } else {
       record[node.name] = value;
     }
   }
 }
+
+function isDefined<T>(val: T | undefined): val is T {
+  return val !== undefined;
+}
```

Comment:
Maybe have a validation folder to hold these type of functions as typeguards now that we use typescript